aries_cloudagent/config/argparse.py

```diff
@@ -1605,6 +1615,15 @@ def get_settings(self, args: Namespace):
             elif args.endorser_protocol_role == ENDORSER_ENDORSER:
                 settings["endorser.endorser"] = True
 
+        if args.endorser_invitation:
```

Comment:
Is the `endorser_invitation` meant to supplant `endorser_public_did`?

Reply:
It looks like you can use either: when using the invitation you would not have to manually create the connection between the two agents, though, as it would be done autmatically.

Reply:
In that case the `on_startup_event` function would fail if the invitation is not set but the Endorser's public DID is.

Reply:
The invitation is used to auto-establish the connection between the author and endorser.

The endorser DID is used to set the meta-data on the endpoint (the author needs to know the endorser DID - this has to be part of the transaction that the author signs before sending it to be endorsed).

The on_startup method will establish the connection if there is an invitation, and then set the connection meta-data if it can.  I don't think this will *fail* if the DID isn't provided, however the endorsement protocol can't be used until this DID is set.

I think I'll set the argparse validation to enforce that the endorser DID and alias *must* be provided if the invitation is provided.